pkg/health/aggregation_policy.go

```diff
@@ -16,26 +16,40 @@
 
 package health
 
+import "github.com/InVisionApp/go-health"
+
 // DefaultAggregationPolicy aggregates the healths by constructing a new Health based on the given
 // where the overall health status is negative if one of the healths is negative and positive if all are positive
 type DefaultAggregationPolicy struct {
 }
 
 // Apply aggregates the given healths
-func (*DefaultAggregationPolicy) Apply(healths map[string]*Health) *Health {
+func (*DefaultAggregationPolicy) Apply(healths map[string]health.State, failureTreshold int64) *Health {
 	if len(healths) == 0 {
 		return New().WithDetail("error", "no health indicators registered").Unknown()
 	}
 	overallStatus := StatusUp
 	for _, health := range healths {
-		if health.Status == StatusDown {
+		if health.Status == "failed" && health.ContiguousFailures > failureTreshold {
```

Comment:
`health.ContiguousFailures > failureTreshold` introduces behavior, which I'm not sure we want for all health indicators.
If the storage is down when healthcheck call comes, you might get a response with body that says `failed`, but you'll get status `200 OK`. This will happen if the time between the detection of storage down and the call is less than `failureThreshold * interval`. 
Assuming we configure `interval=60s` and `/v1/monitor/health` is called every 5 minutes - on the 4th minute the storage fails and now on the health call we get `200 OK` but with failure response. Assuming there is a storage outage for 4 minutes (1 minute before the next health call) then the next health call will report UP again. However, the service has not been operational for 4 minutes and all storage operations were failing.

Does it make sense to have the health configuration be a `map[string]Settings` where each health indicator can be separately configured with `failureThreshold` and `interval`? Then we might configure the storage healthcheck to report down even after 1 failure.